### PR TITLE
Use `AnonymizedFaker` for primary keys

### LINKED ISSUE #1807: Contextual Anonymization transformers shouldn't be used for primary keys
### Environment Details

* SDV version: 1.10.0 (+ SDV Enterprise 0.10.0)
* Python version: Any
* Operating System: Any

### Error Description
My metadata might list a semantic, PII type as a primary key -- for example an `email`.

```json
{
  "primary_key": "user_email",
  "columns": {
    "user_email": { "sdtype": "email" },
    ...
  }
}
```

If I am an SDV Enterprise user _and_ my primary keys are of premium sdtypes (`email`, `phone_number`, ...), then SDV will assign the primary key column to a contextual anonymization transformer such as [DomainBasedAnonymizer](https://docs.sdv.dev/rdt/transformers-glossary/deep-data-understanding/email/domainbasedanonymizer) or [AnonymizedGeoExtractor](https://docs.sdv.dev/rdt/transformers-glossary/deep-data-understanding/phone-number/anonymizedgeoextractor). **Such transformers are not designed to keep uniqueness in mind**, meaning that it may be possible that some primary keys repeat.

### Solution
As a quick solution, we should ensure that we do not assign Contextual Anonymization transformers to primary keys. Instead, we should fall back on AnonymizedFaker like we do for public SDV.

(In the future, we can think about how Contextual Anonymization transformers can support uniqueness. We'd likely need to think through a different algorithm for each one.)

### PR DESCRIPTION
**Pull request by fealho**
CU-86azdzv1n, Resolve #1807.

### CODE REVIEW

**Comment by sdv-team** [comment]
Task linked: [CU-86azdzv1n SDV - Contextual Anonymization transformers shouldn't be used for primary keys #1807](https://app.clickup.com/t/86azdzv1n)

**Comment by fealho** [comment]
Results of the change:
Screenshot 2024-03-07 at 10 06 33 AM (2)

**Review by frances-h — COMMENTED** [review]
Could we add a test for this as well?

**Comment by fealho** [comment]
> Could we add a test for this as well?

I can't write a test which actually verifies the logic, since that requires SDV-Enterprise. I can add a unit test that runs through the logic though.

File: tests/unit/data_processing/test_data_processor.py [hunk 1]
```diff
@@ -1496,6 +1496,40 @@ def test__create_config_with_address_columns(self):
         }
         assert config == expected_config
 
+    def test__create_config_with_contextual_column_as_primary_key(self):
+        """Test it when a conextual column is a primary key."""
```
Comment (frances-h) [inline tests/unit/data_processing/test_data_processor.py:1500]: typo: conextual -> contextual

**Review by frances-h — APPROVED** [review]
looks good, thanks for addressing!

**Review by R-Palazzo — APPROVED** [review]

**Pull request merged**